src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Scanner;
+
+public class Duke {
+    private static Task[] taskList = new Task[100];
+    private static int taskNum = 0;
+
+    private static void listTasks() {
+        for (int i = 0; i < taskNum; i++) {
+            Task task = taskList[i];
+            System.out.println((i+1) + ". " + task);
+        }
+    }
+
+    /**
+     * Process user input and do corresponding task specified.
+     *
+     * @param command   the user input
+     * @return          true if the program should exit, false if otherwise
+     */
+    private static boolean processCommand(String command) throws DukeException {
+        boolean shouldExit = false;
+        command = command.strip();  // ensure commands don't get affected by extra spaces
+
```

Comment:
i agree that this if else is quite long, but it's still readable, smaller functions might help even more? :)